x-pack/plugins/index_management/__jest__/components/index_table.test.js

```diff
@@ -76,39 +80,54 @@ const status = (rendered, row = 0) => {
 const snapshot = (rendered) => {
   expect(rendered).toMatchSnapshot();
 };
+
 const openMenuAndClickButton = (rendered, rowIndex, buttonIndex) => {
+  // Select a row.
   const checkboxes = findTestSubject(rendered, 'indexTableRowCheckbox');
   checkboxes.at(rowIndex).simulate('change', { target: { checked: true } });
   rendered.update();
+
+  // Click the bulk actions button to open the context menu.
   const actionButton = findTestSubject(rendered, 'indexActionsContextMenuButton');
   actionButton.simulate('click');
   rendered.update();
+
+  // Click an action in the context menu.
   const contextMenuButtons = findTestSubject(rendered, 'indexTableContextMenuButton');
   contextMenuButtons.at(buttonIndex).simulate('click');
+  rendered.update();
 };
-const testEditor = (buttonIndex, rowIndex = 0) => {
-  const rendered = mountWithIntl(component);
+
+const testEditor = (rendered, buttonIndex, rowIndex = 0) => {
   openMenuAndClickButton(rendered, rowIndex, buttonIndex);
   rendered.update();
   snapshot(findTestSubject(rendered, 'detailPanelTabSelected').text());
 };
-const testAction = (buttonIndex, done, rowIndex = 0) => {
-  const rendered = mountWithIntl(component);
-  let count = 0;
+
+const testAction = (rendered, buttonIndex, rowIndex = 0) => {
+  // This is leaking some implementation details about how Redux works. Not sure exactly what's going on
+  // but it looks like we're aware of how many Redux actions are dispatched in response to user interaction,
+  // so we "time" our assertion based on how many Redux actions we observe. This is brittle because it
+  // depends upon how our UI is architected, which will affect how many actions are dispatched.
+  // Expect this to break when we rearchitect the UI.
```

Comment:
I'd rather put: "Expect to completely rewrite those tests when we rearchitect the UI"! 😄 This would be one of our tech debt. Actually I'd first rewrite the tests and follow our CIT best practices (no testing of implementation details!) and only after I would rearchitect the UI.

Reply:
Haha yes. I'm just marking the landmine with this comment, not prescribing how it should be addressed. I think an issue is a better place for that (more discoverable and collaborative).